### DISCUSSION TITLE
Trigger Date Roll Using Calendar

### DISCUSSION THREAD

**Discussion by dmazeall** [quartznet/quartznet#1031] [Q&A]
Hello,

Is there a way to add date rolling to a trigger's schedule using a calendar?

i.e.  I have a trigger scheduled to fire on the first business day of each month.  if the calculated month's first business day falls on a date in a calendar then I would like to the trigger's fire date/time to roll to the next valid business day as opposed to not firing at all.  is this possible?

thanks

-dm

**Comment by lahma** [comment]
For me it sounds that it might be easier calculate valid days beforehand and having daily trigger run against the calendar. As next business day might not be Monday, leap years etc.
